lib/compilers/manyclangs.ts

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) 2022, Compiler Explorer Authors
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are met:
+//
+//     * Redistributions of source code must retain the above copyright notice,
+//       this list of conditions and the following disclaimer.
+//     * Redistributions in binary form must reproduce the above copyright
+//       notice, this list of conditions and the following disclaimer in the
+//       documentation and/or other materials provided with the distribution.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+
+import path from 'path';
+
+import {CompilerInfo} from '../../types/compiler.interfaces';
+import {BaseCompiler} from '../base-compiler';
+
+/**
+ * The manyclangs cmpiler library.
+ */
+export class ManyClangsCompiler extends BaseCompiler {
+    static get key() {
+        return 'manyclangs';
+    }
+
+    constructor(compilerInfo: CompilerInfo & Record<string, any>, env: any) {
+        super(compilerInfo, env);
+        this.compiler.isCompilerLibrary = true;
+    }
+
+    /**
+     * Returns the available clang revisions.
+     */
+    getRevisions() {
+        // TODO
```

Comment:
:+1:
do we have a sense how many there will be in the final setup? will this be a query to search do we think, or will this list be in-RAM in all instances?

Reply:
There will be _a lot_, in the range of ~100k. I've updated the TODO, but my general idea is to take the output of `elfshaker list`, cache it on disk, then read off that "cache" when this list is needed.